src/lazy/binary/encoded_value.rs

```diff
@@ -0,0 +1,268 @@
+use crate::binary::non_blocking::type_descriptor::Header;
+use crate::types::SymbolId;
+use crate::IonType;
+use std::ops::Range;
+
+/// Type, offset, and length information about the serialized value over which the
+/// NonBlockingRawBinaryReader is currently positioned.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct EncodedValue {
```

Comment:
> I was puzzled at first because I think this is not really a Value, per se, but a kind of Descriptor maybe because it doesn't actually hold the value itself (encoded or otherwise). Is that right?

I had a tough time naming this. It represents a value _in the input stream_. Maybe `UnparsedValue`? `ValueOffsets`? Open to other thoughts here, though this is not part of the public API so we can change it down the road too.

> `pub(crate)`

Done!